WooCommerce/Classes/ViewRelated/Products/Variations/ProductsTabProductViewModel+ProductVariation.swift

```diff
@@ -2,42 +2,50 @@ import UIKit
 import Yosemite
 
 extension ProductsTabProductViewModel {
-    init(productVariationModel: EditableProductVariationModel, currency: String) {
+    init(productVariationModel: EditableProductVariationModel, currency: String, currencySettings: CurrencySettings = .shared) {
         imageUrl = productVariationModel.productVariation.image?.src
         name = productVariationModel.name
-        detailsAttributedString = productVariationModel.productVariation.createDetailsAttributedString(currency: currency)
+        detailsAttributedString = productVariationModel.createDetailsAttributedString(currency: currency, currencySettings: currencySettings)
 
         imageService = ServiceLocator.imageService
         isSelected = false
     }
 }
 
-private extension ProductVariation {
-    func createDetailsAttributedString(currency: String) -> NSAttributedString {
-        let visibilityText = createVisibilityText()
-        let priceText = createPriceText(currency: currency)
+private extension EditableProductVariationModel {
+    func createDetailsAttributedString(currency: String, currencySettings: CurrencySettings) -> NSAttributedString {
+        let detailsText: String
+        let textColor: UIColor
 
-        let detailsText = [visibilityText, priceText]
-            .compactMap({ $0 })
-            .joined(separator: " • ")
+        if isEnabled == false {
+            detailsText = DetailsLocalization.disabledText
+            textColor = .textSubtle
+        } else if isEnabledAndMissingPrice {
+            detailsText = DetailsLocalization.noPriceText
+            textColor = .warning
+        } else {
+            detailsText = createPriceText(currency: currency, currencySettings: currencySettings)
+            textColor = .textSubtle
+        }
 
         let attributedString = NSMutableAttributedString(string: detailsText,
                                                          attributes: [
-                                                            .foregroundColor: UIColor.textSubtle,
+                                                            .foregroundColor: textColor,
                                                             .font: StyleManager.footerLabelFont
             ])
         return attributedString
     }
 
-    func createVisibilityText() -> String? {
-        guard purchasable == false else {
-            return nil
-        }
-        return NSLocalizedString("Hidden", comment: "Shown in a Product Variation cell if the variation is not visible (purchasable is false)")
+    func createPriceText(currency: String, currencySettings: CurrencySettings) -> String {
+        let currencyFormatter = CurrencyFormatter(currencySettings: currencySettings)
+        return currencyFormatter.formatAmount(regularPrice ?? productVariation.price, with: currency) ?? ""
```

Comment:
⚠️ Just wanted to double-check this. The previous code only used `price` but now we changed it to `regularPrice`.  Is this intentional? I wasn't sure how this change relates to the other changes in this PR.

Reply:
actually it's a great question that I should have confirmed with G, I updated `price` to `regularPrice` because `price` is a calculated property from the API and it could be showing either the regular price or sale price depending on whether the sale is ongoing at current time. During dev, I found it a bit confusing seeing the sale price on the variation list while we show the regular price first in the price row. However, since G is AFK this week lemme revert back to the previous behavior and I'll confirm with her after she's back. Updated in: https://github.com/woocommerce/woocommerce-ios/pull/2627/commits/d55051f23385bc5241aa70b492b34d3b9b3a9314

Reply:
Thanks for that explanation!